doc/docker.md

```diff
@@ -0,0 +1,189 @@
+# Docker info
+
+The main readme contains the basic steps to start with Docker:
+
+    git clone git@github.com:concord-consortium/rigse.git portal
+    cd portal
+    docker-compose up # this will take 15 minutes to download gems
+
+After this you can browse to [http://0.0.0.0:3000](http://0.0.0.0:3000). On OS X this
+might take more than 5 minutes to load the first page. Look in the terminal where you ran
+`docker-compose up` to monitor progress.
+
+## Notes on basic steps
+
+If you have a clean checkout of the portal, `docker-compose up` should just work. If you
+have another service running on port 3000 you might get a port conflict.
+
+By default your local portal folder will be mounted inside of the docker app service.
+Any changes you make to your local folder will show up immediately in the docker app
+service. The files `config/database.yml`, `config/settings.yml` and
+`config/app_environment_variables.yml` are automatically copied from their `.sample.yml`
+counterparts when you run `docker-compose up` if they do not exist. If they already
+exist, they will not be updated. If they already exist then there is a good chance
+they will not be configured correctly for Docker. You can delete these files and run
+`docker-compose restart` which will fix things.  
+
+Also, when `config/database.yml` is not present yet, `docker-compose up` will copy it
+and run `rake db:setup`. `rake db:setup` erases any data in your database, so be careful.
```

Comment:
Yeah this issue with copying is complex. Whether the files are there after you delete local copies will depend on several factors. I did make a change a little bit ago to make it a bit more deterministic.  Previously the files were added to the docker image built by Docker-dev.  I took that step out.  I'll test these instructions, and make sure I include all of the command needed to remove the files, so `docker-compose up`, does the right setup.

Alternatively we could try to change where Rails looks for these files, then we wouldn't need to copy them. However I think that might be confusing in a different way. 

Ideally we can remove the need for most of these files. Removing app_environment_variables would be easy. And it wouldn't take too much work to remove settings.yml.